### PR TITLE
Issue #94 - improve how groups name is imported

### LINKED ISSUE #94: Options | Improve how groups names are imported
By default, the extension only updates names of groups. If you add more groups (e.g. you currently only have 3, but the file has 4) the new ones won't be added; and if you remove groups on the files (e.g. you have 4, but the file has 3), they won't be removed from the extension.

However, if `overrideAll` is true, the groups should be completely overridden.

> Besides implementing this change (which will probably be a single line change: change `if (_.isEmpty(settings.providersGroups)) {` to `if (overrideAll || _.isEmpty(settings.providersGroups)) {`), also create a unit tests to verify this new behavior.

### PR DESCRIPTION
**Pull request by Saidmurod1996**
Fixes #94

### CODE REVIEW

**Comment by tim-kodirov** [comment]
If I edit groups with "Edit manually" button, the groups are overridden properly. However, if I try to import a file, the groups are gone even if there are some:

image

**Comment by Saidmurod1996** [comment]
@tim-kodirov The bug you found was occurring because we don't allow group length to be less than 3, that is why it was acting weirdly. Now it's fixed, and should work in the following way. So recheck considering the case.

If a file has less than 3 groups you should: update the name of the groups are than defined, and disabled the other groups (without changing their names).
> So if the file only has 1 group, then you will rename Group 1, disable Group 2 and 3, and delete Group 4 to N (if defined).
> And if the file has 2 groups, then rename Group 1 and Group 2, disabled Group 3, and delete Groups 4 to N (if any)

**Comment by Saidmurod1996** [comment]
Regarding 
> Tests are running successfully, but I think that the test only covers "Edit manually" function and doesn't cover "Import" function

In the above "Edit manually" also shouldn't work, you probably checked that with not less than 3 groups. I've also added above cases to unit test so re run tests just in case.

**Comment by tim-kodirov** [comment]
LGTM

File: tests/unit/configFile.spec.js [hunk 1]
```diff
@@ -430,6 +430,49 @@ describe("ConfigFile", () => {
...
+      );
+      settings = (await LocalStore.getOne(StoreKey.SETTINGS)) || {};
+      expectedGroups = ConfigFile.parseGroups(newGroups.slice(0, 4));
+      expect(settings.providersGroups).toEqual(expectedGroups);
+
+      // If new groups are less than 3:
+      // 1) only group name should be updated with new one
+      // 2) disable other groups up to 3
+      // 3) delete all groups greater than 3
+      await ConfigFile.parseJSONFile(
```
Comment (jfmdev) [inline tests/unit/configFile.spec.js:464]: Just one more change: please separate unit tests for each one of these edge cases (e.g. new groups > current groups; new groups < current groups; new groups < 3; new groups = current groups), and group them all using another describe.

Something like:

```js
describe("parseJSONFile()", () => {
    describe("Groups names should be updated is forced (no matter flag's value)", () => {
         it("Groups names should be updated") // newGroups.length == currentGroups.length

         it("New groups should be added") // newGroups.length > currentGroups.length

         it("Old obsolete groups should be removed") // newGroups.length < currentGroups.length

         it("Undefined groups should be disabled") // newGroups.length < 3

```

**Review by jfmdev — CHANGES_REQUESTED** [review]
